src/Arcus.Security.Core/ISecretStore.cs

```diff
@@ -51,12 +46,13 @@ public interface ISecretStore
         /// <param name="name">The name that was used to register the <see cref="ICachedSecretProvider"/> in the secret store.</param>
         /// <typeparam name="TCachedSecretProvider">The concrete <see cref="ICachedSecretProvider"/> type.</typeparam>
         /// <exception cref="ArgumentException">Thrown when the <paramref name="name"/> is blank.</exception>
-        /// <exception cref="KeyNotFoundException">
-        ///     Thrown when there was no <see cref="ICachedSecretProvider"/> found in the secret store with the given <paramref name="name"/>,
-        ///     or there were multiple <see cref="ICachedSecretProvider"/> instances registered with the same name.
+        /// <exception cref="KeyNotFoundException">Thrown when there was no <see cref="ICachedSecretProvider"/> found in the secret store with the given <paramref name="name"/>.</exception>
+        /// <exception cref="NotSupportedException">
+        ///     Thrown when their was either none of the registered secret providers are registered as <see cref="ICachedSecretProvider"/> instances
+        ///     or there was an <see cref="ISecretProvider"/> registered but not with caching.
         /// </exception>
-        /// <exception cref="NotSupportedException">Thrown when there was an <see cref="ICachedSecretProvider"/> registered but not with caching.</exception>
         /// <exception cref="InvalidCastException">Thrown when the registered <see cref="ICachedSecretProvider"/> cannot be cast to the specific <typeparamref name="TCachedSecretProvider"/>.</exception>
+        /// <exception cref="InvalidOperationException">Thrown when multiple <see cref="ICachedSecretProvider"/> were registered with the same name.</exception>
         TCachedSecretProvider GetCachedProvider<TCachedSecretProvider>(string name) where TCachedSecretProvider : ICachedSecretProvider;
```

Comment:
Why do we need the explicit `GetCachedProvider` method ?  Is the consumer who wants to retrieve a secretprovider interested in the fact if the secret provider is cached or not ?   In what situation would one prefer to call this method instead of the more general `GetProvider` method ?

Reply:
The reason here why we have both the 'general' `.GetProvider` and the 'concrete' `.GetProvider<TScretProvider>`; is because people may want to use specific operations on the concrete implementation. The Azure Key Vault secret provider has for example the possibility to store secrets too. But that functionality can't be accessed through the 'general' `ISecretProvider` interface. So we will have to cast that.

In case of a cached version, the same. When people implement their own version, they may want to access specific functionality from that implementation that's not available in the `ISecretProvider`.